### PR TITLE
Use native crierrors package for registry unavailable

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
After vendoring the latest Kubernetes `master`, we're now able to use the CRI errors package without having a need to pin OTEL any more.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: server/image_pull.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (s *Server) PullImage(ctx context.Context, req *types.PullImageRequest) (*t
 
 	if pullOp.err != nil {
 		if errors.Is(pullOp.err, syscall.ECONNREFUSED) {
-			return nil, k8simages.ErrRegistryUnavailable
+			return nil, crierrors.ErrRegistryUnavailable
```
Comment (sohankunkerkar) [inline server/image_pull.go:99]: not a blocker, but the PR would be easy to read if you push this change in a separate commit.

Comment (haircommander) [reply server/image_pull.go:99]: +1 I agree

Comment (saschagrunert) [reply server/image_pull.go:99]: Done, split-up the commits.

**Pull request merged**